shared/js/content-scripts/click-to-load.js

```diff
@@ -644,14 +755,63 @@
+            const textButton = makeTextButton(widgetData.replaceSettings.buttonText, widget.getMode())
+            const { contentBlock: placeholder, shadowRoot } = createContentBlock(
+                widget, button, textButton, icon
+            )
+            button.addEventListener('click', widget.clickFunction(originalElement, placeholder))
+            textButton.addEventListener('click', widget.clickFunction(originalElement, placeholder))
+
+            // Size the placeholder element to match the original video
+            // element.
+            // Note: If the website later resizes the video element, the
+            //       placeholder will not resize to match.
+            const { width, height } = window.getComputedStyle(originalElement)
+            placeholder.style.width = width
+            placeholder.style.height = height
+
+            replaceYouTubeVideoWithPlaceholder(entity, widget, originalElement, placeholder)
+
+            // Show an unblock link if parent element forces small height
+            // which may hide video.
+            // FIXME - This is flawed, vertical position inside parent element is not considered,
+            //         which matters if parent hides overflow.
+            const { height: parentHeight } = window.getComputedStyle(parent)
+            if (parseInt(height, 10) <= 200 || parseInt(parentHeight, 10) <= 200) {
+                const textButton = shadowRoot.querySelector(`#${titleID + 'TextButton'}`)
+                textButton.style.display = 'block'
+            }
+        }
     }
 
-    function replaceClickToLoadElements (config) {
+    /**
+     * Replace the blocked CTP elements on the page with placeholders.
+     * @param {Object} config
+     *   A single entry from the CTP configuration file.
+     * @param {Element} [targetElement]
+     *   If specified, only this element will be replaced (assuming it matches
+     *   one of the expected CSS selectors). If omitted, all matching elements
+     *   in the document will be replaced instead.
+     */
+    async function replaceClickToLoadElements (config, targetElement) {
         for (const entity of Object.keys(config)) {
             for (const widget of Object.values(config[entity].elementData)) {
-                const els = document.querySelectorAll(widget.selectors.join())
-                for (const el of els) {
-                    createReplacementWidget(entity, widget, el)
+                const selector = widget.selectors.join()
+                let elements = []
+
+                if (targetElement) {
+                    if (targetElement.matches(selector)) {
+                        elements.push(targetElement)
+                    }
+                } else {
+                    elements = document.querySelectorAll(widget.selectors.join())
+                }
+                for (const element of elements) {
+                    await createReplacementWidget(entity, widget, element)
                 }
```

Comment:
Perhaps something like:
```
await Promise.all(elements.map((element) => createReplacementWidget(entity, widget, element)))
```
Which should will run them in parallel.

Reply:
I'd prefer to leave this alone, for now at least. I haven't changed that logic so far, so it seems like a kind of unrelated change.

Reply:
The suggestion wasn't to modify the method at all.

The proposed change makes the browser await after all of the calls have happened. I'm happy to make it a follow up clean up task instead; I know we discussed changing out some of the calls in this method previously.

Reply:
Done.